### PR TITLE
fix(BDM): Update the BDM doc to include new API methods

### PR DESCRIPTION
**Pull request by danila-m**
Closes [RUNTIME-295](https://bonitasoft.atlassian.net/browse/RUNTIME-295)

### CODE REVIEW

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc [hunk 1]
```diff
@@ -142,6 +142,8 @@ first if a BDM already exists.
 (and from all the process classloaders of the tenant). *Does not* update or remove anything from the database.
 * https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#cleanAndUninstallBusinessDataModel--[cleanAndUninstallBusinessDataModel()]: same as above plus
 *drops all the BDM tables* resulting in a complete data loss. *Not for production*. Use with caution.
+* https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#updateBusinessDataModel--[updateBusinessDataModel()] Updates the BDM. Uninstalls the old BDM if there was one and installs the new BDM.
```
Comment (pabloalonso) [inline modules/ROOT/pages/how-a-bdm-is-deployed.adoc]: If I am not wrong it cover as well the case that there is no BDM (fresh install). Should we update the installBDM description?

**Review by pabloalonso — CHANGES_REQUESTED** [review]
"Deploying a new version of the BDM on top of an existing BDM needs to uninstall the previous version first."
--> I think we should rephrase this as it has sense before when we were forced to do 2 different calls.

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc [hunk 2]
```diff
@@ -172,8 +174,10 @@ Using the Bonita Portal to deploy a BDM follows this flow:
 * you must be logged in as the tenant administrator
 * go to the BPM services page, and pause the services
 * go to the Business Data Model page, select the BDM zip file to deploy
-* if a BDM already exists on the current tenant, the portal implicitly calls uninstallBusinessDataModel() on the Engine API
-* then the portal calls installBusinessDataModel() on the Engine API, which generates the Java model, loads it in the tenant
+* the portal will call the method updateBusinessDataModel() to install the new BDM.
+This method implicitly calls the two following methods in a single database transaction:
```
Comment (pabloalonso) [inline modules/ROOT/pages/how-a-bdm-is-deployed.adoc]:
```suggestion
This method implicitly perform the following steps in a single database transaction:
```

Comment (pabloalonso) [reply modules/ROOT/pages/how-a-bdm-is-deployed.adoc]: I am not sure about the "single database transaction"
why database?

**Review by pabloalonso — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc (hunk 2 above)
Comment (educhastenier) [reply modules/ROOT/pages/how-a-bdm-is-deployed.adoc]: ~perform~ => perform**s**

Comment (educhastenier) [reply modules/ROOT/pages/how-a-bdm-is-deployed.adoc]:
```suggestion
This method implicitly performs the following steps in a single database transaction:
```

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc (hunk 1 above)
Comment (danila-m) [reply modules/ROOT/pages/how-a-bdm-is-deployed.adoc]: Will do. That being said, the install method still exists in the API, and the limitations on it still apply.

Comment (danila-m) [reply modules/ROOT/pages/how-a-bdm-is-deployed.adoc]: Honestly, I feel we should depreciate the install method.

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc [hunk 3]
```diff
@@ -137,7 +137,9 @@ However, the SQL instructions run by the DBA must precisely match the Java class
 The `TenantAdministrationAPI` exposes several methods:
 
 * https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#installBusinessDataModel-byte:A-[installBusinessDataModel()] that installs a new BDM. Fails if one already exists. Call uninstallBusinessDataModel()
-first if a BDM already exists.
+first if a BDM already exists. Deprecated in 2021.2, as UpdateBusinessDataModel() covers the same use case, and should be used instead.
```
Comment (educhastenier) [inline modules/ROOT/pages/how-a-bdm-is-deployed.adoc]:
```suggestion
first if a BDM already exists. Deprecated in 2021.2, as updateBusinessDataModel() covers the same use case, and should be used instead.
```

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc [hunk 4]
```diff
@@ -137,7 +137,9 @@ However, the SQL instructions run by the DBA must precisely match the Java class
...
+* https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#updateBusinessDataModel--[updateBusinessDataModel()] Updates the BDM. Uninstalls the old BDM if there was one and installs the new BDM.
```
Comment (educhastenier) [inline modules/ROOT/pages/how-a-bdm-is-deployed.adoc]:
```suggestion
* https://javadoc.bonitasoft.com/api/{javadocVersion}/org/bonitasoft/engine/api/TenantAdministrationAPI.html#updateBusinessDataModel--[updateBusinessDataModel()] that updates the BDM. It uninstalls the previous BDM if applicable, and installs the new one.
```

File: modules/ROOT/pages/how-a-bdm-is-deployed.adoc [hunk 5]
```diff
@@ -172,8 +173,10 @@ Using the Bonita Portal to deploy a BDM follows this flow:
 * you must be logged in as the tenant administrator
 * go to the BPM services page, and pause the services
 * go to the Business Data Model page, select the BDM zip file to deploy
-* if a BDM already exists on the current tenant, the portal implicitly calls uninstallBusinessDataModel() on the Engine API
-* then the portal calls installBusinessDataModel() on the Engine API, which generates the Java model, loads it in the tenant
+* the portal will call the method updateBusinessDataModel() to install the new BDM.
+This method implicitly performs the following steps in a single database transaction:
+** if a BDM already exists on the current tenant, uninstallBusinessDataModel() is called.
+** then the portal calls installBusinessDataModel() on the Engine API, which generates the Java model, loads it in the tenant
```
Comment (educhastenier) [inline modules/ROOT/pages/how-a-bdm-is-deployed.adoc]:
```suggestion
** if a BDM already exists on the current tenant, it is uninstalled
** the new BDM is installed / deployed
** the Bonita Engine loads the new BDM in the tenant
```
